cli/azd/pkg/environment/deployment_scope.go

```diff
@@ -4,10 +4,11 @@ type DeploymentScope struct {
 	subscriptionId    string
 	resourceGroupName string
 	resourceName      string
+	resourceType      string
 }
 
-func NewDeploymentScope(subscriptionId string, resourceGroupName string, resourceName string) *DeploymentScope {
-	return &DeploymentScope{subscriptionId: subscriptionId, resourceGroupName: resourceGroupName, resourceName: resourceName}
+func NewDeploymentScope(subscriptionId string, resourceGroupName string, resourceName string, resourceType string) *DeploymentScope {
```

Comment:
I wonder if at some point we should change the name of this thing to something like `TargetResource`?  It feels like at some level basically this is just a nice wrapper around a resource ID that allows you to pull out structured information from the ID.

I think the word "Deployment" really futzes with my head here, since I always think of "Deployment" to mean "an arm deployment" instead of "an app deployment".

Feel free to take no action here - just throwing the thought out there.

Reply:
Had the same thought about `TargetResource` being a better name too. Glad that we agree on it, I'll make a change.